debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java

```diff
@@ -119,17 +123,19 @@ else if (connectorConfig.getSnapshotLockingMode() == SnapshotLockingMode.EXCLUSI
                 }
             }
         }
-        else if (connectorConfig.getSnapshotLockingMode() == SnapshotLockingMode.SNAPSHOT) {
-            ((SqlServerSnapshotContext)snapshotContext).preSchemaSnapshotSavepoint = jdbcConnection.connection().setSavepoint("dbz_schema_snapshot");
-        }
         else {
             throw new IllegalStateException("Unknown locking mode specified.");
         }
     }
 
     @Override
     protected void releaseSchemaSnapshotLocks(SnapshotContext snapshotContext) throws SQLException {
-        jdbcConnection.connection().rollback(((SqlServerSnapshotContext)snapshotContext).preSchemaSnapshotSavepoint);
+        // Exclusive mode: locks should be kept until the end of transaction.
+        // read_uncommitted mode; snapshot mode: no locks have been acquired.
```

Comment:
I am not sure it makes sense. Snapshot level is specific.
One can encounter an error as below:
```
Snapshot isolation transaction failed in database 'X' because the object accessed by the statement has been modified by a DDL statement in another concurrent transaction since the start of this transaction. It is disallowed because the metadata is not versioned. A concurrent update to metadata can lead to inconsistency if mixed with snapshot isolation.
```
As far as I understand, in this case taking locks for schema snapshot does not help in any way.

I believe well-documented restriction for this level should be enough.

Reply:
Ok; so is this to say that any concurrent DDL op will fail the snapshot essentially?

Reply:
I am not sure that any concurrent DDL will cause failure, but for sure any DDL executed on tables being fetched.